Source/Particles/Deposition/ChargeDeposition.H

```diff
@@ -90,22 +114,31 @@ void doChargeDepositionShapeN (const GetParticlePosition& GetPosition,
     }
 #endif
 
+    const auto dxiarr = geom.InvCellSizeArray();
+    const auto plo = geom.ProbLoArray();
+    const auto domain = geom.Domain();
+
+    const amrex::Box sample_tbox(IntVect(AMREX_D_DECL(0, 0, 0)), a_tbox_max_size - 1);
+    amrex::Box sample_tbox_x = convert(sample_tbox, ix_type);
+
+    sample_tbox_x.grow(depos_order);
+
+    const auto npts = sample_tbox_x.numPts();
+
 #if defined(AMREX_USE_GPU)
-    const int threads_per_block = 128;
-    const int nblocks = (std::max(np_to_deposit, rho_box.numPts()) + threads_per_block - 1) / threads_per_block;
-    std::size_t shared_mem_bytes = rho_box.numPts()*sizeof(amrex::Real);
+    const int nblocks = a_bins.numBins();
+    const auto offsets_ptr = a_bins.offsetsPtr();
+    const int threads_per_block = 256;
+
+    std::size_t shared_mem_bytes = npts*sizeof(amrex::Real);
+    amrex::IntVect bin_size = WarpX::sort_bin_size;
 
     const std::size_t max_shared_mem_bytes = amrex::Gpu::Device::sharedMemPerBlock();
 
     if (shared_mem_bytes > max_shared_mem_bytes) {
       std::cout << "Tile size for charge deposition of " << shared_mem_bytes
                 << " bytes would not fit in GPU's shared memory of " << max_shared_mem_bytes << " bytes" << std::endl;
-#if 1
-      std::cout << "Falling back to direct global memory access" << std::endl;
-      shared_mem_bytes = 0;
-#else
       amrex::Abort("Tile size too big for GPU shared memory charge deposition");
```

Comment:
This case gets hit at the first step after initialization, with obviously garbage huge values:

`Tile size for charge deposition of 52377931632 bytes would not fit in GPU's shared memory of 49152 bytes`

Reply:
Andrew's fixes handled this.